oauth2/handler.go

```diff
@@ -576,7 +576,16 @@ func (h *Handler) TokenHandler(w http.ResponseWriter, r *http.Request) {
 		session.DefaultSession.Claims.Issuer = strings.TrimRight(h.c.IssuerURL().String(), "/") + "/"
 		session.DefaultSession.Claims.IssuedAt = time.Now().UTC()
 
-		for _, scope := range accessRequest.GetRequestedScopes() {
+		var scopes = accessRequest.GetRequestedScopes()
+
+		// Added for compatibility with MITREid
+		if h.c.IncludeAllScopesWhenEmpty() && len(scopes) == 0 {
```

Comment:
I just realized that this does not have a test yet. Could you please add a test to ensure that this feature works as intended? I'm happy to help if you need help.

Reply:
I'm not sure how to implement the test, I'm pretty new working with go and I'm not sure how to test it. 

could you guide me?

Reply:
Sure, you could add an option [here](https://github.com/ory/hydra/blob/master/oauth2/oauth2_client_credentials_test.go#L82) to indicate if the the scope should be included or not. Then you add a new test [after here](https://github.com/ory/hydra/blob/master/oauth2/oauth2_client_credentials_test.go#L171) (you can probably copy/paste a success case from earlier and remove unnecessary duplication). In the test, you would send an empty scope and expect the resulting access token to have the full scope however. You should add another test where the opposite is the case - so `h.c.IncludeAllScopesWhenEmpty` is false and then you expect the scope to be empty. You could also add another test to check if this behaves correctly when `len(scopes) != 0`. Hope this helps!

To run the tests, you need docker installed locally and then run `make test` in the project root directory.

Reply:
@aeneasr  I've created the tests